src/CSharp/CodeCracker/Usage/MethodAnalyzers/MethodChecker.cs

```
@@ -0,0 +1,106 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CodeCracker.Usage.MethodAnalyzers
+{
+    public class MethodChecker
+    {
+        private readonly SyntaxNodeAnalysisContext context;
+        private readonly DiagnosticDescriptor diagnosticDescriptor;
+
+        public MethodChecker(SyntaxNodeAnalysisContext context, DiagnosticDescriptor diagnosticDescriptor)
+        {
+            this.context = context;
+            this.diagnosticDescriptor = diagnosticDescriptor;
+        }
+
+        public void AnalyzeConstrutor(MethodInformation methodInformation)
+        {
+            if (ConstructorNameNotFound(methodInformation) || MethodFullNameNotFound(methodInformation.MethodFullDefinition))
+            {
+                return;
+            }
+            var argumentList = ((ObjectCreationExpressionSyntax)context.Node).ArgumentList;
+            var arguments = GetArguments(argumentList);
+            Execute(methodInformation, arguments, argumentList);
+        }
+
+        private bool ConstructorNameNotFound(MethodInformation methodInformation)
+        {
+            return AbreviatedConstructorNameNotFound(methodInformation) && QualifiedConstructorNameNotFound(methodInformation);
+        }
+
+        private bool AbreviatedConstructorNameNotFound(MethodInformation methodInformation)
+        {
+            var objectCreationExpressionSyntax = (ObjectCreationExpressionSyntax) context.Node;
+            var identifier = objectCreationExpressionSyntax.Type as IdentifierNameSyntax;
+            return identifier?.Identifier.ValueText != methodInformation.MethodName;
+        }
+
+        private bool QualifiedConstructorNameNotFound(MethodInformation methodInformation)
+        {
+            var objectCreationExpressionSyntax = (ObjectCreationExpressionSyntax)context.Node;
+            var identifier = objectCreationExpressionSyntax.Type as QualifiedNameSyntax;
+            return identifier?.Right.ToString() != methodInformation.MethodName;
```

Yes, it could, but it will failed once the semantic model is loaded (MethodFullNameNotFound).

The idea is to fail fast without loading the semantic model. I don't think there's a way to do that only with the syntax tree, because you could name your namespace System.Net as well.

Do you know a way? I'll investigate this further.